squad_3_ad_data_science/main.py

```diff
@@ -25,6 +40,48 @@ def features(**kwargs):
     """
     print("==> GENERATING DATASETS FOR TRAINING YOUR MODEL")
 
+    # As we do not count on obtaining new dataset dinamically, this function
+    # just checks if the expected datasets are present on the right folder
+    if not isfile(config.train_dataset):
+        logger.error(f'Train dataset `{config.train_dataset}` not found.' +
+                     'Model cannot be trained.')
+
+        exit(1)
+
+    else:
+        logger.info(f'Train dataset `{config.train_dataset}` found!')
+
+    for test_ds in config.test_datasets:
+        if not isfile(test_ds):
+            logger.error(f'Test dataset `{config.train_dataset}` not found.' +
+                         'Model may not be tested. Fix or remove from tests.')
+
+        else:
+            logger.info(f'Test dataset `{config.train_dataset}` found!')
+
+    logger.info('Loading data')
+    pure_data = pd.read_csv(config.train_dataset, index_col=0)
+
+    old_shape = pure_data.shape
+    logger.info('Working on features')
+    try:
+        data = pipeline_data(pure_data)
+    except Exception:
+        logger.error('Something where wrong')
```

Comment:
"something went wrong"

Reply:
Oops. Forget about it. Previous errors log should clarify what happened. I will rewrite this log too.